sdk/servicebus/service-bus/src/serializers/queueResourceSerializer.ts

```diff
@@ -310,11 +320,101 @@ export interface InternalQueueOptions {
 /**
  * Represents all attributes of a queue entity
  */
-export interface QueueDetails extends QueueOptions {
+export interface QueueDetails {
   /**
    * Name of the queue
    */
-  queueName?: string;
+  queueName: string;
+
+  /**
+   * Determines the amount of time in seconds in which a message should be locked for processing by a receiver. After this period, the message is unlocked and available for consumption by the next receiver. Settable only at queue creation time.
+   */
+  lockDuration: string;
+
+  /**
+   * The entity's size in bytes.
+   *
+   */
+  sizeInBytes: number;
+
+  /**
+   * Specifies the maximum queue size in megabytes. Any attempt to enqueue a message that will cause the queue to exceed this value will fail.
+   */
+  maxSizeInMegabytes: number;
+
+  /**
+   * The entity's message count.
+   *
+   */
+  messageCount: number;
+
+  /**
+   * Depending on whether DeadLettering is enabled, a message is automatically moved to the DeadLetterQueue or deleted if it has been stored in the queue for longer than the specified time. This value is overwritten by a TTL specified on the message if and only if the message TTL is smaller than the TTL set on the queue. This value is immutable after the Queue has been created.
+   */
+  defaultMessageTimeToLive: string;
+
+  /**
+   * Specifies the time span during which the Service Bus detects message duplication.
+   */
+  duplicateDetectionHistoryTimeWindow: string;
+
+  /**
+   * Entity to forward deadlettered messages to
+   *
+   */
+  forwardDeadLetteredMessagesTo?: string;
+
+  /**
+   * Max idle time before entity is deleted
```

Comment:
Minor: Can you go through and make sure these comments have proper punctuation?  (i.e. ends with periods)

Reply:
Hm, it's difficult to look for just the changes statements. Would rather so a general one pass at dotting in the end or a diff PR so that things don't get too noisy

Reply:
I think you can do it as part of this PR.

Reply:
Also, is it just me, or does this help text not match the property description? What is this property for? What's an example value?

Reply:
Yes, was looking to discuss this as well in https://github.com/Azure/azure-sdk-for-js/issues/5858#issuecomment-547559715

It supposedly deletes the entity if it is idle for the set duration. Default value is "P10675199DT2H48M5.4775807S"
We could rename to `maxWaitTimeForAutoDelete`? The internal option will have to continue being "autoDeleteOnIdle"

Reply:
Let's not rename properties. If the service team ever documents their API, then we should have a 1:1 mapping between the words we use on the property and the names they use.

@ramya0820 The only actionable item in this comment thread is the one that @chradek made on punctuation.

I believe updating the docs for the property with example value is being done in #6044.
Do you want to consider folding in those changes into this PR since the types are getting split here?

Reply:
Can you clarify what do you mean by "folding"?
Yes, #6044 is blocked on this PR being merged as discussed earlier. Can add the docs as part of it once this is in on master.

Reply:
By folding, I meant doing the changes that are part of that PR in this PR because we are doing documentation updates here as well. Also because the discussion above touched examples which is getting covered in #6044

Reply:
Ok, looks like we are going forward with renaming of properties instead - did we want to consider `autoDeleteOnIdleMaxTime` or `autoDeleteOnMaxIdleTime` so we don't entirely deviate from service's naming convention?